### ISSUE TITLE
Scribbles - Implement a likelihood model using Gaussian Mixture Model

### ISSUE THREAD

**Issue by masadcv** [Project-MONAI/MONAILabel#439]
We currently have a histogram-based likelihood model for estimating appearance-based proposals for foreground and background.
While this works for most of the sample apps, it may be more relevant to implement and include more advanced likelihood models (see some discussion here: https://github.com/Project-MONAI/MONAILabel/issues/432#issuecomment-932309545)

For example, the original GrabCut paper (https://www.microsoft.com/en-us/research/wp-content/uploads/2004/08/siggraph04-grabcut.pdf) uses a Gaussian Mixture Model (GMMs) which may be useful and a bit more advanced than histograms

There are a number of implementations for GMMs out there, perhaps the most relevant ones will be the GPU implementation from MONAI and scikit-learn's CPU implementation:
- MONAI (GPU): https://docs.monai.io/en/latest/networks.html?highlight=gmm#monai.networks.layers.GaussianMixtureModel 
- Scikit-learn (CPU): https://scikit-learn.org/stable/modules/generated/sklearn.mixture.GaussianMixture.html

GMM can also naturally extend to multi-class problems, so it will be relevant for when we extend scribbles for multi-class.

**Comment by tvercaut** [comment]
The underpinning mixture model feature probably belongs to MONAI rather than MONAI Label. That said, it makes sense to have it accessible for MONAI Label.

Related pointers:
- https://github.com/Project-MONAI/MONAI/issues/1601
- https://github.com/ldeecke/gmm-torch
- https://pytorch.org/docs/stable/distributions.html#mixturesamefamily

By the way, this might be interesting to represent the histogram-based likelihood:
https://pytorch.org/docs/stable/distributions.html#categorical

**Comment by masadcv** [comment]
Thanks @tvercaut - I think most of the transforms in ./monailabel/scribbles/transform.py belong to MONAI. They are kept here for updating/improving as much as possible before they are moved. 

In regard to GMM in MONAILabel, how I see it for now is that this will primarily be a transform that uses the GMM from MONAI. Once this is working as expected and looks in final shape, we can move it to MONAI. 

Once we have GMM, GrabCut may be doable in MONAILabel. I am aware that this may be achieving something similar to the GrabCut issue in MONAI. Has there been any update on that after GMM?

Thanks for sharing the pytorch distribution function. I will check it out in the coming days.

**Comment by masadcv** [comment]
@SachidanandAlle resolving this issue will give us a new infer method for scribbles. It may potentially address some limitations of histogram method, especially in ambiguous cases. 

Do you want me to include this alongside Histogram+GraphCut method we have in radiology app?

**Comment by SachidanandAlle** [comment]
Makes sense.. already some were complaining about empty results for certain images.. 
Also multiple options will help others to follow and contribute more such methods

**Comment by masadcv** [comment]
Okay many thanks! I am looking into this now and will add this once it is ready.
